administrator/components/com_cookiemanager/src/Model/ConsentsModel.php

```diff
@@ -0,0 +1,149 @@
+	 */
+	public function __construct($config = array())
+	{
+		if (empty($config['filter_fields']))
+		{
+			$config['filter_fields'] = array(
+				'id', 'a.id',
+				'uuid', 'a.uuid',
+				'ccuuid', 'a.ccuuid',
+				'consent_opt_in', 'a.consent_opt_in',
+				'consent_opt_out', 'a.consent_opt_out',
+				'consent_date', 'a.consent_date',
+				'user_agent', 'a.user_agent',
+				'url', 'a.url',
+			);
+		}
+
+		parent::__construct($config);
+	}
+
+	/**
+	 * Method to auto-populate the model state.
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @param   string  $ordering   An optional ordering field.
+	 * @param   string  $direction  An optional direction (asc|desc).
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function populateState($ordering = 'a.id', $direction = 'desc')
+	{
+		$search = $this->getUserStateFromRequest($this->context . 'filter.search', 'filter_search');
+		$this->setState('filter.search', $search);
+
+		// List state information.
+		parent::populateState($ordering, $direction);
+	}
+
+	/**
+	 * Method to get a store id based on model configuration state.
+	 *
+	 * This is necessary because the model is used by the component and
+	 * different modules that might need different sets of data or different
+	 * ordering requirements.
+	 *
+	 * @param   string  $id  A prefix for the store id.
+	 *
+	 * @return  string  A store id.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getStoreId($id = '')
+	{
+		// Compile the store id.
+		$id .= ':' . $this->getState('filter.search');
+
+		return parent::getStoreId($id);
+	}
+
+	/**
+	 * Build an SQL query to load the list data.
+	 *
+	 * @return  \Joomla\Database\DatabaseQuery
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getListQuery()
+	{
+		// Create a new query object.
+		$db    = $this->getDbo();
+		$query = $db->getQuery(true);
+
+			$query->select(
+				$this->getState(
+					'list.select',
+					'a.id, a.uuid, a.ccuuid, a.consent_opt_in, a.consent_opt_out, a.consent_date, a.user_agent, a.url'
+				)
+			);
+			$query->from($db->quoteName('#__cookiemanager_consents', 'a'));
+
+		// Filter by search in ccuuid.
+		$search = $this->getState('filter.search');
+
+		if (!empty($search))
+		{
+			$search = $db->quote('%' . str_replace(' ', '%', $db->escape(trim($search), true) . '%'));
+			$query->where('(a.ccuuid LIKE ' . $search . ')');
+		}
+
+		// Add the list ordering clause.
+		$orderCol  = $this->state->get('list.ordering', 'a.id');
+		$orderDirn = $this->state->get('list.direction', 'DESC');
+
+		$query->order($db->escape($orderCol) . ' ' . $db->escape($orderDirn));
+
+		return $query;
+	}
+
+	/**
+	 * Method to get a list of consents.
+	 *
+	 * @return  mixed  An array of data items on success, false on failure.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getItems()
+	{
+		$items = parent::getItems();
+
+		return $items;

```

Comment:
Unnecessary local variable

```suggestion
		return parent::getItems();
```